Microsoft.Toolkit.Win32.UI.Controls/WPF/WebView/WebView.cs

```diff
@@ -0,0 +1,1046 @@
+using Microsoft.Toolkit.Win32.UI.Controls.WinForms;
+using Point = Windows.Foundation.Point;
+using Size = Windows.Foundation.Size;
+
+/*
+
+Overview of Keyboard Input Routing for IWebViewControl
+
+The window hosting IWebViewControl receives regular alphanumeric keyboard input via its WndProc, and whoever is running the message loop
+calls DispatchMessage() after any preprocessing and special routing, and the message gets to the WndProc of the window with focus. "Accelerator"
+keys need to be passed to the IWebViewControl via TranslateAccelerator()
+
+However, due to the way the window is hosted out of process, we are not able to intercept those messages. As such we have no support to handle
+tabbing in or out of the IWebBrowserControl window.
+
+NOTE: Depending on how the control is hosted, input messages flow different
+
+ */
+
+namespace Microsoft.Toolkit.Win32.UI.Controls.WPF
+{
+    /// <summary>
+    /// This class is an implementation of <see cref="IWebView"/> for WPF. This class cannot be inherited.
+    /// </summary>
+    /// <seealso cref="Microsoft.Toolkit.Win32.UI.Controls.WPF.WebViewHost" />
+    /// <seealso cref="Microsoft.Toolkit.Win32.UI.Controls.IWebView" />
+    public sealed class WebView : WebViewHost, IWebView
+    {
+        private static readonly Hashtable InvalidatorMap = new Hashtable();
+
+        private delegate void PropertyInvalidator(WebView webViewHost);
+
+        private static readonly DependencyProperty IsIndexedDBEnabledProperty = DependencyProperty.Register(
+            nameof(IsIndexedDBEnabled),
+            typeof(bool),
+            typeof(WebView),
+            new PropertyMetadata(true, PropertyChangedCallback));
+
+        private static readonly DependencyProperty IsJavaScriptEnabledProperty = DependencyProperty.Register(
+            nameof(IsJavaScriptEnabled),
+            typeof(bool),
+            typeof(WebView),
+            new PropertyMetadata(true, PropertyChangedCallback));
+
+        private static readonly DependencyProperty IsScriptNotifyAllowedProperty = DependencyProperty.Register(
+            nameof(IsScriptNotifyAllowed),
+            typeof(bool),
+            typeof(WebView),
+            new PropertyMetadata(true, PropertyChangedCallback));
+
+        private static readonly bool IsWebPermissionRestricted = !Security.CallerAndAppDomainHaveUnrestrictedWebBrowserPermission();
+
+        private static readonly DependencyProperty SourceProperty = DependencyProperty.Register(
+            nameof(Source),
+            typeof(Uri),
+            typeof(WebView),
+            new PropertyMetadata(new Uri("about:blank"), PropertyChangedCallback));
+
+        private WebViewControlProcess _process;
+        private WebViewControlHost _webViewControl;
+
+        [SecuritySafeCritical]
+        [SuppressMessage("Microsoft.Design", "CA1065")]
+        static WebView()
+        {
+#pragma warning disable 1065
+            if (IsWebPermissionRestricted)
```

Comment:
@normesta @mcleanbyron A series of conditions here where we can fail. Details in comments